backend/metamodel/syson-sysml-metamodel/src/main/java/org/eclipse/syson/sysml/impl/ElementImpl.java

```diff
@@ -530,46 +530,44 @@ else if (eNotificationRequired())
 
     /**
      * <!-- begin-user-doc -->
+     * The full ownership-qualified name of this Element, represented in a form that is valid according to the KerML
+     * textual concrete syntax for qualified names (including use of unrestricted name notation and escaped characters, as
+     * necessary). The qualifiedName is null if this Element has no owningNamespace or if there is not a complete
+     * ownership chain of named Namespaces from a root Namespace to this Element.
      * <!-- end-user-doc -->
      * @generated NOT
      */
-    @Override
-    public String getQualifiedName() {
-        boolean qualifiedNameContainsNull = false;
-        StringBuilder qualifiedNameBuilder = new StringBuilder();
-        EObject container = this.eContainer();
-        if (container instanceof Membership membership) {
-            EObject membershipContainer = membership.eContainer();
-            if (membershipContainer instanceof Element element) {
-                String elementQN = element.getQualifiedName();
-                if (elementQN == null) {
-                    qualifiedNameContainsNull = true;
-                } else {
-                    qualifiedNameBuilder.append(elementQN);
-                    qualifiedNameBuilder.append("::");
-                }
-            }
-        } else if (container instanceof Element element) {
-            String elementQN = element.getQualifiedName();
-            if (elementQN == null) {
-                qualifiedNameContainsNull = true;
-            } else {
-                qualifiedNameBuilder.append(elementQN);
-                qualifiedNameBuilder.append("::");
-            }
-        }
-        String name = this.getName();
-        if (name == null || name.isBlank()) {
-            qualifiedNameContainsNull = true;
-        } else if (name.contains("\s")) {
-            name = "'" + name + "'";
-        }
-        qualifiedNameBuilder.append(name);
-        if (qualifiedNameContainsNull) {
-            return null;
-        }
-        return qualifiedNameBuilder.toString();
-    }
+     @Override
+     public String getQualifiedName() {
+         String selfName = this.getName();
+         if (selfName == null || selfName.isBlank()) {
+             return null;
+         }
+ 
```

Comment:
Here you should call _org.eclipse.syson.sysml.export.utils.Appender.toPrintableName(String)_. It handle both space and name starting with digit case

Reply:
Are you sure about calling an export module method from metamodel module?

Reply:
No I asked Guillaume (orally) to move the code into an helper inside the MM module.

Reply:
Done